sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,248 @@
+        # ``relative_velocity`` so choose the first ``ReferenceFrame`` that
+        # ``relative_position`` is defined using.
+        frame = relative_position.args[0][1]
+        relative_velocity = relative_position.diff(t, frame)
+        shortening_velocity = -relative_velocity.dot(relative_position.normalize())
+        return shortening_velocity
+
+    def compute_loads(self, force: Expr) -> list[tuple[Point, Vector]]:
+        """Loads required by the equations of motion method classes.
+
+        Explanation
+        ===========
+
+        ``KanesMethod`` requires a list of ``Point``-``Vector`` tuples to be
+        passed to the ``loads`` parameters of its ``kanes_equations`` method
+        when constructing the equations of motion. This method acts as a
+        utility to produce the correctly-structred pairs of points and vectors
+        required so that these can be easily concatenated with other items in
+        the list of loads and passed to ``KanesMethod.kanes_equations``. These
+        loads are also in the correct form to also be passed to the other
+        equations of motion method classes, e.g. ``LagrangesMethod``.
+
+        Examples
+        ========
+
+        The below example shows how to generate the loads produced in a linear
+        actuator that produces a contractile force ``F``. First, create a
+        linear actuator between two points separated by the coordinate ``q``
+        in the ``x`` direction of the global frame ``N``.
+
+        >>> from sympy.physics.mechanics import Point, ReferenceFrame
+        >>> from sympy.physics.mechanics._pathway import LinearPathway
+        >>> from sympy.physics.vector import dynamicsymbols
+        >>> q = dynamicsymbols('q')
+        >>> N = ReferenceFrame('N')
+        >>> attachments = (Point('pO'), Point('pI'))
+        >>> attachments[-1].set_pos(attachments[0], q * N.x)
+        >>> linear_pathway = LinearPathway(attachments)
+
+        Now create a symbol ``F`` to describe the magnitude of the
+        (contractile) force that will be produced along the pathway. The list
+        of loads that ``KanesMethod`` requires can be produced by calling the
+        pathway's ``compute_loads`` method with ``F`` passed as the only
+        argument.
+
+        >>> from sympy import Symbol
+        >>> F = Symbol('F')
+        >>> linear_pathway.compute_loads(F)
+        [(pO, F*q(t)/sqrt(q(t)**2)*N.x), (pI, - F*q(t)/sqrt(q(t)**2)*N.x)]
+
+        Parameters
+        ==========
+
+        force : Expr
+            The force acting along the length of the pathway. It is assumed
+            that this ``Expr`` represents a contractile force.
+
+        """
+        relative_position = self.attachments[-1].pos_from(self.attachments[0])
+        loads = [
+            (self.attachments[0], force * relative_position / self.length),
+            (self.attachments[-1], -force * relative_position / self.length),
```

Comment:
This math needs to be in the docs. So the force is set to force*change in position/original position ? Do all "Pathways" create forces of this nature?

Reply:
Also, is this relationship only of interest in muscle modeling?

Reply:
This is of interest to all pathways. It's producing the `(Point, Vector)` pairs to be passed as loads to `KanesMethod` or `LagrangesMethod`. `force` is just some scalar value. For example, if the pathway is representing that of a linear spring then the value passed to this method will be `spring_constant * self.length`. The other part (`relative_position / self.length`) is a unit vector specifying the direction in which that force acts. So for `self.attachments[0]`, this is a unit vector in the direction of `self.attachments[-1]` from `self.attachments[0]`.

This is simple for a linear pathway, but when I add obstacle-set pathways (pathway's with via points) and pathways wrapping surfaces this will become considerably more useful and `loads` will often contain more than two entries.

Reply:
Looking again to this, makes me wonder whether I even agree at all, with this. It assumes that an actuator will depend on the relative position, which is incorrect.

Reply:
> Looking again to this, makes me wonder whether I even agree at all, with having this function, called `compute_loads`. It assumes a certain implementation of an actuator, namely that the actuator force depends linearly on the position. While this is true for springs, which are approximated as being linear, it is not true for a lot of other mechanical actuators.

This is incorrect because the `force` argument need not be scalar, it can easily be a function of pathway length and pathway shortening velocity. `force` specifies the magnitude of the force while the vector component of the tuples determines the directions in which that force acts. This implementation will not require any assumptions be made about the force characteristics of the actuator.

Reply:
Oh wait, I've read it wrong based on one of the earlier comments. You just specify the direction of the force here. There is no assumption here that the force would be dependent on the path length.

Reply:
> Oh wait, I've read it wrong based on one of the earlier comments. You just specify the direction of the force here. There is no assumption here that the force would be dependent on the path length.

Exactly 🙂

Also, once https://github.com/sympy/sympy/pull/24924 goes in this should be refactored to use `Force` and `Torque`.